### ISSUE TITLE
Documentation Playgrounds

### ISSUE THREAD

**Issue by BasThomas** [Moya/Moya#1097]
It would be really cool if we could create a Playground / Playground Book to guide new users through the library. Doing this via Playgrounds allows us to easily guide the user along the process with live examples and inline comments.

Is this something people would like to see?

**Comment by sunshinejr** [comment]
Good idea. I really like the way RxSwift did its [Rx.playground](https://github.com/ReactiveX/RxSwift/tree/master/Rx.playground). We could use our documentation & examples to create some chapters (initial idea): Setup -> Basic usage -> Closures -> Stubbing -> Tests -> Plugins -> Advanced usage.

But then, having in mind we wanted to make a website(#1079), having also docs with examples, we would have to find a way of maintaining it all, or choose something we actually can maintain 🤔

**Comment by SD10** [comment]
No promises or set time frame but I'd like to take a stab at this. I think a Playground driven approach would result in a more fine-grained and chronological learning experience. Compile time errors might even make it easier to maintain than static documentation 😅. So if anyone has something to add to the discussion feel free to let me know.

I think the concern regarding maintenance is insightful and agree it would be difficult to maintain a large amount of content. I also agree that if the community decides to go both directions, then we should find a way to reuse as much documentation as possible.

**Comment by BasThomas** [comment]
That'd be awesome, @SD10!

On maintaining everything: yeah, this will definitely be some more work. But as @SD10 brought up, having "compiling documentation" makes it easier to spot errors and fix them.

I am also still not sure about the website, but let me comment about that in its ticket.

**Comment by SD10** [comment]
Hey there everyone, this is something I'd like to move forward on. I've started a branch here:
https://github.com/SD10/Moya/tree/playground

Basically, I took the [Targets.md](https://github.com/Moya/Moya/blob/master/docs/Targets.md) documentation and converted it to a playground file. I'm trying to re-use as much as the existing docs as possible, however, if we're looking for an authentic walkthrough I can do that too.

I think reusing as much existing documentation as possible will make things easier to maintain. I also like the idea of having our static documentation backed up by a compiled playground. This will immediately trigger errors for out of date documentation.

I'd just like some feedback on my approach before continuing. One thing I'd like to know is if the wording in the Playground file should mirror the Documentation exactly. My current implementation paraphrased the existing documentation.

Thanks for any feedback!

**Comment by sunshinejr** [comment]
This looks cool, @SD10! Made two comments in there, other than that I love it. I think there should be a fine line between reusing existing docs and making the walkthrough feel more like a learning process.

**Comment by SD10** [comment]
@sunshinejr I replied to your suggestions. Thank you, I really appreciate the feedback.

**Comment by sunshinejr** [comment]
Not sure if you are still working on this one @SD10 - might be worth adding a label for help needed if not 😉

**Comment by SD10** [comment]
A hacktoberfest label would be great. I gave up on this because I wasn't satisfied with how it was turning out. Something I may revisit in the future if no one else picks it up before me 😄

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by sunshinejr** [comment]
Cool, I've added the label @SD10.
